src/zenml/integrations/langchain/materializers/vector_store_materializer.py

```diff
@@ -0,0 +1,59 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the langchain vector store materializer."""
+
+
+import os
+import pickle
+from typing import Type
+
+from langchain.vectorstores import VectorStore
+
+from zenml.enums import ArtifactType
+from zenml.io import fileio
+from zenml.materializers.base_materializer import BaseMaterializer
+
+DEFAULT_FILENAME = "vectorstore.pkl"
+
+
+class LangchainVectorStoreMaterializer(BaseMaterializer):
+
+    ASSOCIATED_ARTIFACT_TYPE = ArtifactType.DATA
+    ASSOCIATED_TYPES = (VectorStore,)
+
+    def load(self, data_type: Type[VectorStore]) -> VectorStore:
+        """Reads a langchain vector store from a pickle file.
+
+        Args:
+            data_type: The type of the vector store.
+
+        Returns:
+            The vector store.
+        """
+        super().load(data_type)
+        filepath = os.path.join(self.uri, DEFAULT_FILENAME)
+        with fileio.open(filepath, "rb") as fid:
+            vector_store = pickle.load(fid)
```

Comment:
same as above. really basic checks in a utility function might help. maybe put everything behind a zenml.pickle function that checks metadata from file stored in an artifact store

Reply:
Validation has been added in now.